tests/unit_tests/constraint/test_psifcialg.py

```diff
@@ -0,0 +1,298 @@
+        oracle = Oracle(G)
+        alg = PsiFCI(known_intervention_targets=False, ci_estimator=oracle, cd_estimator=oracle)
+
+        self.context_func = lambda: make_context(create_using=InterventionalContextBuilder)
+        self.G = G
+        self.ci_estimator = oracle
+        self.alg = alg
+
+    def test_figure2_skeleton(self):
+        """Test learning the graph for Figure 2 in :footcite:`Jaber2020causal`."""
+        # first create the oracle
+        directed_edges = [
+            ("x", "w"),
+            ("x", "y"),
+            ("y", "w"),
+            ("z", "y"),
+            ("z", "x"),
+        ]
+        bidirected_edges = [("x", "w")]
+        graph = pgraphs.AugmentedGraph(
+            incoming_directed_edges=directed_edges, incoming_bidirected_edges=bidirected_edges
+        )
+        non_f_graph = graph.copy()
+        graph.add_f_node({"x", "w"})
+        oracle = Oracle(graph)
+
+        # define the expected graph we will learn
+        edges = [
+            (("F", 0), "x"),
+            (("F", 0), "w"),
+            ("x", "w"),
+            ("x", "z"),
+            ("x", "y"),
+            ("z", "y"),
+            ("z", "w"),
+            ("w", "y"),
+        ]
+        expected_skeleton = nx.Graph(edges)
+        obs_expected_skeleton = expected_skeleton.copy()
+        obs_expected_skeleton.remove_node(("F", 0))
+
+        # define the learner and the context
+        learner = PsiFCI(ci_estimator=oracle, cd_estimator=oracle, known_intervention_targets=False)
+        data = [dummy_sample(non_f_graph), dummy_sample(non_f_graph)]
+        context = (
+            make_context(create_using=InterventionalContextBuilder)
+            .variables(data=data[0])
+            .num_distributions(2)
+            .build()
+        )
+        learner.fit(data, context)
+
+        # first check the observational skeleton
+        skel_graph = learner.skeleton_learner_.adj_graph_
+        obs_skel_graph = learner.skeleton_learner_.context_.state_variable("obs_skel_graph")
+
+        assert nx.is_isomorphic(obs_expected_skeleton, obs_skel_graph, edge_match=None)
+        assert nx.is_isomorphic(expected_skeleton, skel_graph)
+
+        # now check the end graph
+        directed_edges = [
+            (("F", 0), "x"),
+            (("F", 0), "w"),
+            ("z", "x"),
+            ("z", "y"),
+            ("z", "w"),
+            ("x", "y"),
+            ("x", "w"),
+            ("y", "w"),
+        ]
+        circle_edges = [("x", "z"), ("w", "x"), ("w", "z"), ("w", "y")]
+        expected_G = PsiPAG(
+            incoming_directed_edges=directed_edges, incoming_circle_edges=circle_edges
+        )
+        expected_G.graph["F-nodes"][("F", 0)] = ["x"]
+
+        learned_graph = learner.graph_
+        for edge_type, subgraph in expected_G.get_graphs().items():
+            assert nx.is_isomorphic(subgraph, learned_graph.get_graphs(edge_type))
+
+
+@pytest.mark.skip()
+def test_psifci_withsachs():
```

Comment:
Gonna remove this / refactor into an oracle-based test, or quicker version once the example runs nicely.